src/wallet/wallet.cpp

```diff
@@ -1147,6 +1153,50 @@ void CWallet::BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) {
 
 
 
+void CWallet::BlockUntilSyncedToCurrentChain() {
```

Comment:
I'm not sure how it is not thread safe? It blocks the *current* thread, not any other work.